### PR TITLE
Fix: loot item charges

### LINKED ISSUE #211: Loot: Charged amulets and charged rings
Charged amulets and charged rings are coming with 1 charge from the monsters.

elder wyrm
{id = "shockwave amulet", chance = 100},
{id = "lightning pendant", chance = 520},

after: 
{id = "shockwave amulet", charges = 5, chance = 100},
{id = "lightning pendant", charges = 200, chance = 520},

### LINKED ISSUE #212: Showing Charges.
In tibia RL, every item with charges, show the real amount.
but in server just showing one charge.
Examples.
23:55 You see a silver amulet (protection earth +10%) that has 200 charges left.
It weighs 5.00 oz.
IN RL
but in last branch have problems with         <attribute key="showcharges" value="1" />

23:56 You see a silver amulet (protection earth +10%) that has 1 charge left.
It weighs 5.00 oz.  all items with charges just showing one charge

### LINKED ISSUE #1883: Showing Charges.
In tibia RL, every item with charges, show the real amount.
but in server just showing one charge.
Examples.
23:55 You see a silver amulet (protection earth +10%) that has 200 charges left.
It weighs 5.00 oz.
IN RL
but in last branch have problems with         <attribute key="showcharges" value="1" />

23:56 You see a silver amulet (protection earth +10%) that has 1 charge left.
It weighs 5.00 oz.  all items with charges just showing one charge

### LINKED ISSUE #1943: Loot: Charged amulets and charged rings
Charged amulets and charged rings are coming with 1 charge from the monsters.

elder wyrm
{id = "shockwave amulet", chance = 100},
{id = "lightning pendant", chance = 520},

after: 
{id = "shockwave amulet", charges = 5, chance = 100},
{id = "lightning pendant", charges = 200, chance = 520},

### PR DESCRIPTION
**Pull request by marcosvf132**
Resolves #1943
Resolves #1883

Set item charges if applicable. Also leave the opportunite to set custom charges/subtype.

**Thanks to @debone and @IgorCFG for the help.**

### CODE REVIEW

File: data/scripts/lib/register_monster_type.lua [hunk 1]
```diff
@@ -217,6 +217,10 @@ registerMonsterType.loot = function(mtype, mask)
 			if loot.aid or loot.actionId then
 				parent:setActionId(loot.aid or loot.actionId)
 			end
+			local lType = ItemType(loot.id)
+			if lType and lType:getCharges() > 1 then
+				parent:setSubType(lType:getCharges())
+			end
 			if loot.subType or loot.charges then
 				parent:setSubType(loot.subType or loot.charges)
 			end
```
Comment (IgorCFG) [inline data/scripts/lib/register_monster_type.lua]: I think you can put it in the `else` the way it is would be overwritten anyway.

```lua
if loot.subType or loot.charges then
    parent:setSubType(loot.subType or loot.charges)
else
    local lType = ItemType(loot.id)
    if lType and lType:getCharges() > 1 then
        parent:setSubType(lType:getCharges())
    end
end
```

..something like that, that way wouldn't create the `lType` in vain

File: data/scripts/lib/register_monster_type.lua [hunk 2]
```diff
@@ -238,6 +242,10 @@ registerMonsterType.loot = function(mtype, mask)
 					if children.aid or children.actionId then
 						child:setActionId(children.aid or children.actionId)
 					end
+					local cType = ItemType(children.id)
```
Comment (IgorCFG) [inline data/scripts/lib/register_monster_type.lua]: the same here

**Review by heberholanda — APPROVED** [review]
Tests, it worked perfectly. Good job Marcos!

**Review by IgorCFG — APPROVED** [review]
🎉🚀

**Review by heberholanda — APPROVED** [review]

**Pull request merged**